server/src/main/java/org/opensearch/extensions/DiscoveryExtension.java

```diff
@@ -61,6 +67,7 @@ public void writeTo(StreamOutput out) throws IOException {
     public DiscoveryExtension(final StreamInput in) throws IOException {
         super(in);
         this.pluginInfo = new PluginInfo(in);
+        this.dependencies = new ArrayList<ExtensionDenpendency>();
```

Comment:
You need the counterpart here of whatever you're writing in `writeTo()` above.  Simplest approach is probably to read the list size, declare your array list with that size, and then iterate up to that size and add to it.  [See example here](https://github.com/opensearch-project/anomaly-detection/blob/a3e2b994ed347263a8dbaf5beed6bd5ddd6931a4/src/main/java/org/opensearch/ad/transport/ADResultBulkRequest.java#L36-L40)

There's another option of `in.readList()` that involves using a `Reader` but a more simple iteration will probably work here.